### PR TITLE
fix: consistently read versions in Ansible

### LINKED ISSUE #2714: Windows: agent logs are not printed
We usually print out the agent logs after a scenario, but not for Windows

cc/ @dikshachauhan-qasource

### PR DESCRIPTION
**Pull request by mdelapenya**
## What does this PR do?
It fixes how the version variables are set in Ansible defining the following criterium:

0. if STACK_VERSION does not exist, use the .stack_version file
1. if the variable exists, use it (i.e. BEAT_VERSION, ELASTIC_AGENT_VERSION, KIBANA_VERSION)
2. if not, use STACK_VERSION
3. if not, use the .stack_version

## Why is it important?
Testing locally for versions that are different to the one in the .stack_version file is not possible, as the version is overwritten

## Checklist

- [x] My code follows the style guidelines of this project
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have made corresponding change to the default configuration files
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have run the Unit tests (`make unit-test`), and they are passing locally
- [ ] I have run the End-2-End tests for the suite I'm working on, and they are passing locally
- [ ] I have noticed new Go dependencies (run `make notice` in the proper directory)

## Author's Checklist

- [ ] Is the Jenkinsfile passing the variables to the stack/test execution? I know the variables are defined at the top-level, but want to make sure they are passed to the enclosing blocks to run the stack node and to run the tests in the test node.

## How to test this PR locally

1. Create a stack
```
make -C .ci create-stack STACK_VERSION="FOOO" BEAT_VERSION="FOOO"
```

The logs should display that the variable will be applied to the `.env` file

## Related issues

- Closes #2714

### CODE REVIEW

**Comment by mdelapenya** [comment]
Ok, calling the build with `FOOO` as the value in the Jenkins UI does pass the values to Ansible:

```
[2022-06-30T11:09:25.402Z] changed: [3.21.55.41] => (item=STACK_VERSION="FOO")
[2022-06-30T11:09:26.342Z] changed: [3.21.55.41] => (item=KIBANA_VERSION="FOO")
[2022-06-30T11:09:27.280Z] changed: [3.21.55.41] => (item=ELASTIC_AGENT_VERSION="FOO")
```

Trigering the build with defaults to see it passing

File: .ci/Jenkinsfile [hunk 1]
```diff
@@ -534,11 +534,14 @@ def generateFunctionalTestStep(Map args = [:]){
...
+  def githubCheckSha1 = params.GITHUB_CHECK_SHA1?.trim() ? params.GITHUB_CHECK_SHA1 : ''
+  def githubCheckRepo = params.GITHUB_CHECK_REPO?.trim() ? params.GITHUB_CHECK_REPO : ''
+
   // Setup environment for platform
   def envContext = []
   envContext.add("PROVIDER=${provider}")
-  envContext.add("GITHUB_CHECK_SHA1=${env.GITHUB_CHECK_SHA1}")
-  envContext.add("GITHUB_CHECK_REPO=${env.GITHUB_CHECK_REPO}")
+  envContext.add("GITHUB_CHECK_SHA1=${githubCheckSha1}")
+  envContext.add("GITHUB_CHECK_REPO=${githubCheckRepo}")
```
Comment (v1v) [inline .ci/Jenkinsfile:544]: IIUC, those env variables were already in place as per:
- https://github.com/elastic/e2e-testing/blob/742ab4bacb3ed53c45cf34a841e636e831ed3fc3/.ci/Jenkinsfile#L86-L87

**Comment by mdelapenya** [comment]
Closes #2713

**Comment by dikshachauhan-qasource** [comment]
Hi @mdelapenya 

Thanks for sharing the required info for running tests on required build versions.

We are now able to create stack on specified build version with help of Stack_version tag.

Attempted to create stack with below versions:
- 8.3.2
- 8.2.1

Thanks!

**Pull request merged**